vault/acme_billing_system_view.go

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package vault
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/base64"
+	"fmt"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/vault/sdk/logical"
+)
+
+type acmeBillingSystemViewImpl struct {
+	extendedSystemView
+	logical.ManagedKeySystemView
+
+	core  *Core
+	entry *MountEntry
+}
+
+var _ logical.ACMEBillingSystemView = (*acmeBillingSystemViewImpl)(nil)
+
+func (c *Core) NewAcmeBillingSystemView(sysView interface{}, managed logical.ManagedKeySystemView) *acmeBillingSystemViewImpl {
+	es := sysView.(extendedSystemViewImpl)
+	des := es.dynamicSystemView
+
+	return &acmeBillingSystemViewImpl{
+		extendedSystemView:   es,
+		ManagedKeySystemView: managed,
+		core:                 c,
+		entry:                des.mountEntry,
+	}
+}
+
+func (a *acmeBillingSystemViewImpl) CreateActivityCountEventForIdentifiers(ctx context.Context, identifiers []string) error {
+	// Fake our clientID from the identifiers, but ensure it is
+	// independent of ordering.
+	//
+	// TODO: Because of prefixing currently handled by AddEventToFragment,
+	// we do not need to ensure it is globally unique.
+	sort.Strings(identifiers)
+	joinedIdentifiers := strings.Join(identifiers, ".")
```

Comment:
this is odd - do we want someone who registers "bar.foo.zip" to look like a duplicate of someone who registers "bar" and "foo.zip"?

Reply:
Hmm, maybe we should reuse `hopeDelim` then.

Reply:
I don't know that this is strictly correct though. 

Note that we force the prefix (`dns/` or `ip/`) onto the identifier in `doTrackBilling` in the `pki/` namespace. This means that the former is:

`dns/bar.foo.zip`, whereas the latter is `dns/bar.dns/foo.zip`, which is a different value.